### PR TITLE
Added Quiz app

### PR DESCRIPTION
**Pull request by JatinRanka**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -10,20 +10,33 @@
...
+    <link
+      rel="stylesheet"
+      href="https://fonts.googleapis.com/icon?family=Material+Icons+Filled"
+    />
 
-      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
-      work correctly both with client-side routing and a non-root public URL.
-      Learn how to configure a non-root public URL by running `npm run build`.
-    -->
     <title>React App</title>
```
Comment (adarshwritescode) [inline public/index.html]: change title to Quiz App :)

File: src/components/QuestionComponent/index.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  userSelectedAnswerIndex: number;
+  currentOptionIndex: number;
+  isCorrectAnswer: boolean;
+};
+
+type QuestionComponentType = {
+  question: Question;
+  currentQuestionIndex: number;
+  totalQuestions: number;
+};
```
Comment (adarshwritescode) [inline src/components/QuestionComponent/index.tsx]: it'd be easy if we move all types In separate file

File: src/components/QuestionComponent/index.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  if (!showAnswers) return style;
+  if (isCorrectAnswer) {
+    style = "success";
+  }
+  if (userSelectedAnswerIndex === currentOptionIndex && !isCorrectAnswer) {
+    style = "danger";
+  }
+
+  return style;
+};
```
Comment (adarshwritescode) [inline src/components/QuestionComponent/index.tsx]: we can move it out in helper | util folder to reuse it anywhere

File: src/components/QuestionComponent/index.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+              className={`option ${optionStyle}`}
+              key={index}
+              onClick={handleOptionSelect}
+              data-index={index}
+            >
+              <p>{option.text}</p>
+            </div>
+          );
+        })}
+      </div>
```
Comment (adarshwritescode) [inline src/components/QuestionComponent/index.tsx:87]: can be separated in OptionList component, it'd be easy to debug and reuse

**Review by WhoAdarshPandya — CHANGES_REQUESTED** [review]
well written code jatin , suggesting few visual changes apart from review

- some padding in navbar elements
- some padding in text fields (mixing with icons)
- enforce user for alphanum password
- change lorem ipsum texts (in quiz description)
- night mode with context will be cherry on the cake 
- make box around quiz questions with some shadow to make it standout
- sort results according to the highest score
- increase timer to 10 sec at least so it'd be easy for user to read question
- change click events from type `any` to appropriate event type

I'm glad you followed all the conventions while coding, your code was well written and most importantly readable :)

**Comment by JatinRanka** [comment]
@WhoAdarshPandya Thank you so much for your detailed review. Your review will surely improve my project a lot.

**Pull request merged**